log/keyvalue_test.go

```diff
@@ -0,0 +1,302 @@
+	t.Run("AsFloat64", testErrKind(v.AsFloat64, "AsFloat64", k))
+	t.Run("AsInt64", testErrKind(v.AsInt64, "AsInt64", k))
+	t.Run("AsString", func(t *testing.T) {
+		assert.Equal(t, val, v.AsString(), "AsString")
+	})
+	t.Run("AsBytes", testErrKind(v.AsBytes, "AsBytes", k))
+	t.Run("AsSlice", testErrKind(v.AsSlice, "AsSlice", k))
+	t.Run("AsMap", testErrKind(v.AsMap, "AsMap", k))
+}
+
+func TestBytes(t *testing.T) {
+	const key = "key"
+	val := []byte{3, 2, 1}
+	kv := log.Bytes(key, val)
+	testKV(t, key, val, kv)
+
+	v, k := kv.Value, log.KindBytes
+	t.Run("AsBool", testErrKind(v.AsBool, "AsBool", k))
+	t.Run("AsFloat64", testErrKind(v.AsFloat64, "AsFloat64", k))
+	t.Run("AsInt64", testErrKind(v.AsInt64, "AsInt64", k))
+	t.Run("AsString", testErrKind(v.AsString, "AsString", k))
+	t.Run("AsBytes", func(t *testing.T) {
+		assert.Equal(t, val, v.AsBytes(), "AsBytes")
+	})
+	t.Run("AsSlice", testErrKind(v.AsSlice, "AsSlice", k))
+	t.Run("AsMap", testErrKind(v.AsMap, "AsMap", k))
+}
+
+func TestSlice(t *testing.T) {
+	const key = "key"
+	val := []log.Value{log.IntValue(3), log.StringValue("foo")}
+	kv := log.Slice(key, val...)
+	testKV(t, key, val, kv)
+
+	v, k := kv.Value, log.KindSlice
+	t.Run("AsBool", testErrKind(v.AsBool, "AsBool", k))
+	t.Run("AsFloat64", testErrKind(v.AsFloat64, "AsFloat64", k))
+	t.Run("AsInt64", testErrKind(v.AsInt64, "AsInt64", k))
+	t.Run("AsString", testErrKind(v.AsString, "AsString", k))
+	t.Run("AsBytes", testErrKind(v.AsBytes, "AsBytes", k))
+	t.Run("AsSlice", func(t *testing.T) {
+		assert.Equal(t, val, v.AsSlice(), "AsSlice")
+	})
+	t.Run("AsMap", testErrKind(v.AsMap, "AsMap", k))
+}
+
+func TestMap(t *testing.T) {
+	const key = "key"
+	val := []log.KeyValue{
+		log.Slice("l", log.IntValue(3), log.StringValue("foo")),
+		log.Bytes("b", []byte{3, 5, 7}),
+	}
+	kv := log.Map(key, val...)
+	testKV(t, key, val, kv)
+
+	v, k := kv.Value, log.KindMap
+	t.Run("AsBool", testErrKind(v.AsBool, "AsBool", k))
+	t.Run("AsFloat64", testErrKind(v.AsFloat64, "AsFloat64", k))
+	t.Run("AsInt64", testErrKind(v.AsInt64, "AsInt64", k))
+	t.Run("AsString", testErrKind(v.AsString, "AsString", k))
+	t.Run("AsBytes", testErrKind(v.AsBytes, "AsBytes", k))
+	t.Run("AsSlice", testErrKind(v.AsSlice, "AsSlice", k))
+	t.Run("AsMap", func(t *testing.T) {
+		assert.Equal(t, val, v.AsMap(), "AsMap")
+	})
+}
+
+type logSink struct {
+	logr.LogSink
+
+	err           error
+	msg           string
+	keysAndValues []interface{}
+}
+
+func (l *logSink) Error(err error, msg string, keysAndValues ...interface{}) {
+	l.err, l.msg, l.keysAndValues = err, msg, keysAndValues
+	l.LogSink.Error(err, msg, keysAndValues)
+}
+
+var stdLogger = stdr.New(golog.New(os.Stderr, "", golog.LstdFlags|golog.Lshortfile))
+
+func testErrKind[T any](f func() T, msg string, k log.Kind) func(*testing.T) {
+	return func(t *testing.T) {
+		l := &logSink{LogSink: testr.New(t).GetSink()}
+		global.SetLogger(logr.New(l))
+		t.Cleanup(func() { global.SetLogger(stdLogger) })
```

Comment:
I think we should export `global.getLogger` to avoid copy-pasting:

```
var stdLogger = stdr.New(golog.New(os.Stderr, "", golog.LstdFlags|golog.Lshortfile))
```


This should be tackled as a separate issue and addressed in other places as well.